src/vs/workbench/services/workspaces/common/workspaceTrust.ts

```diff
@@ -464,4 +497,49 @@ export class WorkspaceTrustTransitionManager extends Disposable {
 	}
 }
 
+class WorkspaceTrustState {
+	private readonly _memento: Memento;
+	private readonly _mementoObject: MementoObject;
+
+	private readonly _acceptsNonWorkspaceFilesKey = 'acceptsNonWorkspaceFiles';
+	private readonly _canSetWorkspaceTrust = 'canSetWorkspaceTrust';
+	private readonly _isTrustedKey = 'isTrusted';
+
+	constructor(storageService: IStorageService) {
+		this._memento = new Memento('workspaceTrust', storageService);
+		this._mementoObject = this._memento.getMemento(StorageScope.WORKSPACE, StorageTarget.MACHINE);
+	}
+
+	get acceptsNonWorkspaceFiles(): boolean {
+		return this._mementoObject[this._acceptsNonWorkspaceFilesKey] ?? false;
+	}
+
+	set acceptsNonWorkspaceFiles(value: boolean) {
+		this._mementoObject[this._acceptsNonWorkspaceFilesKey] = value;
+		this._memento.saveMemento();
+	}
+
+	get canSetWorkspaceTrust(): boolean {
+		return this._mementoObject[this._canSetWorkspaceTrust] ?? true;
+	}
+
+	set canSetWorkspaceTrust(value: boolean) {
```

Comment:
adding this was confusing to me, based on its usage, I interpret this will be for resolvers, but that's not clear to me. also, this doesn't seem to be part of this PR. can we pull this out of this change until its clear how its going to be used.

Reply:
Removed the code.